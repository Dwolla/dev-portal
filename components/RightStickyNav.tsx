import React, { useState } from "react";
import styled from "@emotion/styled";
import {
  GREY_2,
  GREY_5,
  GREY_6,
  HEADLINE_TEXT,
  ORANGE_PRIMARY,
} from "./colors";

const Group = styled.div`
  width: 235px;
  position: sticky;
  top: 0;
  color: ${GREY_6};
  font-family: "Roboto", sans-serif;
  font-size: 13px;
  line-height: 19px;
  cursor: pointer;
`;

const Heading = styled.div`
  height: 16px;
  color: ${GREY_5};
  font-family: "Poppins", sans-serif;
  font-size: 11px;
  font-weight: 500;
  letter-spacing: 0.79px;
  line-height: 17px;
  margin: 0px 0px 7px 1px;
  cursor: text;
`;

const Item = styled.a`
  display: flex;
  width: 221px;
  padding-top: 8px;
  padding-bottom: 8px;
  text-decoration: none;
  box-sizing: border-box;
  border-left: 1px solid ${GREY_2};
  &.tab-1,
  &.tab-2 {
    padding-left: 15px;
  }
  &.tab-3 {
    padding-left: 30px;
  }
  &.tab-4 {
    padding-left: 45px;
  }
  :hover {
    color: ${HEADLINE_TEXT};
  }
  &.active {
    border-left: 1px solid ${ORANGE_PRIMARY};
    color: ${HEADLINE_TEXT};
  }
`;

const TopOfPage = styled.div`
  margin-top: 7px;
  :hover {
    color: ${HEADLINE_TEXT};
  }
`;

type Props = { headers: Record<string, string>[] };

function RightStickyNav(props: Props) {
  const [activeItem, setActiveItem] = useState(null);

  const handleClickActiveItem = (e) => {
    const newActiveItem = e;
    setActiveItem(newActiveItem);
  };

  return (
    <Group>
      <Heading>ON THIS PAGE</Heading>
      {props.headers.map((item) => (
        <Item
          key={item.key}
          className={`tab-${item.level} ${
            activeItem === item.key ? "active" : ""
          }`}
          onClick={() => handleClickActiveItem(item.key)}
        >
          {item.text}
        </Item>
      ))}
      <TopOfPage onClick={() => handleClickActiveItem(null)}>
        Top of Page
      </TopOfPage>
    </Group>
  );
}

export default RightStickyNav;
