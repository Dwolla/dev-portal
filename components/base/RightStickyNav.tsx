import React from "react";
import styled from "@emotion/styled";
import { Link, animateScroll } from "react-scroll";
import {
  GREY_2,
  GREY_5,
  GREY_6,
  HEADLINE_TEXT,
  ORANGE_PRIMARY,
} from "../colors";
import { maxWidth, BREAKPOINT_DESKTOP } from "../breakpoints";
import { ROBOTO, POPPINS } from "../typography";

const { scrollToTop } = animateScroll;

const Group = styled.div`
  @media (${maxWidth(BREAKPOINT_DESKTOP)}) {
    display: none;
  }

  width: 235px;
  margin: auto;
  padding: 20px;
  position: sticky;
  top: 0;

  > a {
    display: flex;
    width: 221px;
    padding-top: 8px;
    padding-bottom: 8px;
    color: ${GREY_6};
    font-family: ${ROBOTO};
    font-size: 13px;
    line-height: 19px;
    text-decoration: none;
    box-sizing: border-box;
    border-left: 1px solid ${GREY_2};
    cursor: pointer;

    &.tab-1,
    &.tab-2 {
      padding-left: 15px;
    }

    &.tab-3 {
      padding-left: 30px;
    }

    &.tab-4 {
      padding-left: 45px;
    }

    :hover {
      color: ${HEADLINE_TEXT};
    }

    &.is-active {
      color: ${HEADLINE_TEXT};
      border-left: 1px solid ${ORANGE_PRIMARY};
    }
  }
`;

const Heading = styled.div`
  @media (${maxWidth(BREAKPOINT_DESKTOP)}) {
    display: none;
  }

  height: 16px;
  color: ${GREY_5};
  font-family: ${POPPINS};
  font-size: 11px;
  font-weight: 500;
  letter-spacing: 0.79px;
  line-height: 17px;
  margin: 0px 0px 7px 1px;
`;

const PageTop = styled.div`
  @media (${maxWidth(BREAKPOINT_DESKTOP)}) {
    display: none;
  }

  color: ${GREY_6};
  font-family: ${ROBOTO};
  font-size: 13px;
  line-height: 19px;
  padding-top: 10px;
  cursor: pointer;

  :hover {
    color: ${HEADLINE_TEXT};
  }
`;

type Heading = { key: string; level: number; title: string };

function RightStickyNav({ headings }: { headings: Heading[] }) {
  return (
    <Group>
      <Heading>ON THIS PAGE</Heading>

      {headings.map((h) => (
        <Link
          key={h.key}
          activeClass="is-active"
          className={`tab-${h.level}`}
          to={h.key}
          spy
          smooth
          offset={0}
          duration={500}
        >
          {h.title}
        </Link>
      ))}

      <PageTop onClick={scrollToTop}>Top of Page</PageTop>
    </Group>
  );
}

export default RightStickyNav;
