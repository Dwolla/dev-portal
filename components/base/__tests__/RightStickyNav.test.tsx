import renderer from "react-test-renderer";
import RightStickyNav from "../RightStickyNav";

// Headings array to pass as props to RightStickyNav
const headings = [
  {
    key: "1",
    level: 1,
    title: "The Basics",
  },
  {
    key: "2",
    level: 2,
    title: "Verification statuses and corresponding events",
  },
  {
    key: "3",
    level: 3,
    title: "Handling retry status",
  },
  {
    key: "4",
    level: 3,
    title: "Handling status: suspended",
  },
  {
    key: "5",
    level: 4,
    title: "Determining verification documents needed",
  },
  {
    key: "6",
    level: 2,
    title: "Level 2 Heading",
  },
  {
    key: "7",
    level: 3,
    title: "Handling retry status",
  },
];

test("RightStickyNav", () => {
  const tree = renderer.create(<RightStickyNav headings={headings} />).toJSON();

  expect(tree).toMatchSnapshot();
});
