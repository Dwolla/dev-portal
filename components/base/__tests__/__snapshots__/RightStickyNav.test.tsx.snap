// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With content 1`] = `
<div>
  <div
    className="css-1iggbn7-SectionStyle egzsi4z0"
  >
    <div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="1"
      >
        <h1>
          The Basics
        </h1>
        <p>
          Paragraph 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id nibh tortor id aliquet. Sed lectus vestibulum mattis ullamcorper velit sed. Nec nam aliquam sem et tortor. Elit eget gravida cum sociis natoque. Ipsum a arcu cursus vitae. Massa sapien faucibus et molestie ac feugiat sed. Turpis egestas maecenas pharetra convallis posuere morbi leo. Duis at consectetur lorem donec massa sapien faucibus et molestie. Purus sit amet luctus venenatis lectus. Pellentesque nec nam aliquam sem et tortor consequat id porta. Sapien et ligula ullamcorper malesuada. Fringilla ut morbi tincidunt augue interdum velit euismod. Vitae turpis massa sed elementum tempus egestas sed. Netus et malesuada fames ac turpis egestas integer eget aliquet. Blandit cursus risus at ultrices mi tempus imperdiet nulla malesuada. Aliquam eleifend mi in nulla posuere sollicitudin aliquam ultrices sagittis. Nisi est sit amet facilisis. Vitae congue eu consequat ac felis. Facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nisl condimentum id venenatis a condimentum. Rutrum tellus pellentesque eu tincidunt tortor. Semper auctor neque vitae tempus quam. Pellentesque adipiscing commodo elit at imperdiet dui accumsan sit. Mi ipsum faucibus vitae aliquet nec ullamcorper sit. Enim lobortis scelerisque fermentum dui faucibus. Volutpat ac tincidunt vitae semper quis lectus nulla at. Pellentesque id nibh tortor id aliquet lectus proin. Blandit cursus risus at ultrices mi. Iaculis eu non diam phasellus vestibulum lorem. Et odio pellentesque diam volutpat commodo. Sodales ut etiam sit amet nisl purus in. Adipiscing elit ut aliquam purus sit. Bibendum enim facilisis gravida neque convallis a cras. Duis at consectetur lorem donec massa sapien faucibus et molestie. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Non odio euismod lacinia at quis risus sed vulputate odio. Vel turpis nunc eget lorem dolor sed viverra ipsum. A lacus vestibulum sed arcu. Odio aenean sed adipiscing diam. Volutpat ac tincidunt vitae semper quis lectus. Tristique magna sit amet purus. Euismod in pellentesque massa placerat duis. Tellus molestie nunc non blandit massa enim nec dui nunc. Faucibus pulvinar elementum integer enim neque volutpat ac tincidunt vitae. Tempus quam pellentesque nec nam aliquam sem et tortor consequat. Ac felis donec et odio pellentesque diam volutpat commodo sed. Orci sagittis eu volutpat odio. Purus faucibus ornare suspendisse sed nisi lacus. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque. Habitant morbi tristique senectus et netus et. Porta non pulvinar neque laoreet suspendisse. Nunc scelerisque viverra mauris in aliquam sem fringilla ut morbi. Tellus integer feugiat scelerisque varius. Pulvinar pellentesque habitant morbi tristique senectus et netus et malesuada. Pharetra et ultrices neque ornare aenean euismod. Proin sagittis nisl rhoncus mattis rhoncus. Nisi quis eleifend quam adipiscing vitae. Eget mauris pharetra et ultrices neque ornare aenean euismod elementum. Non arcu risus quis varius quam quisque id diam. Pharetra magna ac placerat vestibulum. Lobortis feugiat vivamus at augue eget arcu dictum varius duis. Sed id semper risus in hendrerit gravida. Ultrices eros in cursus turpis massa. Non tellus orci ac auctor augue mauris augue. Arcu vitae elementum curabitur vitae nunc. Convallis a cras semper auctor neque vitae tempus quam. Varius quam quisque id diam vel.
        </p>
      </div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="2"
      >
        <h1>
          Verification statuses and corresponding events
        </h1>
        <p>
          Paragraph 2. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id nibh tortor id aliquet. Sed lectus vestibulum mattis ullamcorper velit sed. Nec nam aliquam sem et tortor. Elit eget gravida cum sociis natoque. Ipsum a arcu cursus vitae. Massa sapien faucibus et molestie ac feugiat sed. Turpis egestas maecenas pharetra convallis posuere morbi leo. Duis at consectetur lorem donec massa sapien faucibus et molestie. Purus sit amet luctus venenatis lectus. Pellentesque nec nam aliquam sem et tortor consequat id porta. Sapien et ligula ullamcorper malesuada. Fringilla ut morbi tincidunt augue interdum velit euismod. Vitae turpis massa sed elementum tempus egestas sed. Netus et malesuada fames ac turpis egestas integer eget aliquet. Blandit cursus risus at ultrices mi tempus imperdiet nulla malesuada. Aliquam eleifend mi in nulla posuere sollicitudin aliquam ultrices sagittis. Nisi est sit amet facilisis. Vitae congue eu consequat ac felis. Facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nisl condimentum id venenatis a condimentum. Rutrum tellus pellentesque eu tincidunt tortor. Semper auctor neque vitae tempus quam. Pellentesque adipiscing commodo elit at imperdiet dui accumsan sit. Mi ipsum faucibus vitae aliquet nec ullamcorper sit. Enim lobortis scelerisque fermentum dui faucibus. Volutpat ac tincidunt vitae semper quis lectus nulla at. Pellentesque id nibh tortor id aliquet lectus proin. Blandit cursus risus at ultrices mi. Iaculis eu non diam phasellus vestibulum lorem. Et odio pellentesque diam volutpat commodo. Sodales ut etiam sit amet nisl purus in. Adipiscing elit ut aliquam purus sit. Bibendum enim facilisis gravida neque convallis a cras. Duis at consectetur lorem donec massa sapien faucibus et molestie. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Non odio euismod lacinia at quis risus sed vulputate odio. Vel turpis nunc eget lorem dolor sed viverra ipsum. A lacus vestibulum sed arcu. Odio aenean sed adipiscing diam. Volutpat ac tincidunt vitae semper quis lectus. Tristique magna sit amet purus. Euismod in pellentesque massa placerat duis. Tellus molestie nunc non blandit massa enim nec dui nunc. Faucibus pulvinar elementum integer enim neque volutpat ac tincidunt vitae. Tempus quam pellentesque nec nam aliquam sem et tortor consequat. Ac felis donec et odio pellentesque diam volutpat commodo sed. Orci sagittis eu volutpat odio. Purus faucibus ornare suspendisse sed nisi lacus. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque. Habitant morbi tristique senectus et netus et. Porta non pulvinar neque laoreet suspendisse. Nunc scelerisque viverra mauris in aliquam sem fringilla ut morbi. Tellus integer feugiat scelerisque varius. Pulvinar pellentesque habitant morbi tristique senectus et netus et malesuada. Pharetra et ultrices neque ornare aenean euismod. Proin sagittis nisl rhoncus mattis rhoncus. Nisi quis eleifend quam adipiscing vitae. Eget mauris pharetra et ultrices neque ornare aenean euismod elementum. Non arcu risus quis varius quam quisque id diam. Pharetra magna ac placerat vestibulum. Lobortis feugiat vivamus at augue eget arcu dictum varius duis. Sed id semper risus in hendrerit gravida. Ultrices eros in cursus turpis massa. Non tellus orci ac auctor augue mauris augue. Arcu vitae elementum curabitur vitae nunc. Convallis a cras semper auctor neque vitae tempus quam. Varius quam quisque id diam vel.
        </p>
      </div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="3"
      >
        <h1>
          Handling retry status
        </h1>
        <p>
          Paragraph 3. Paragraph 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id nibh tortor id aliquet. Sed lectus vestibulum mattis ullamcorper velit sed. Nec nam aliquam sem et tortor. Elit eget gravida cum sociis natoque. Ipsum a arcu cursus vitae. Massa sapien faucibus et molestie ac feugiat sed. Turpis egestas maecenas pharetra convallis posuere morbi leo. Duis at consectetur lorem donec massa sapien faucibus et molestie. Purus sit amet luctus venenatis lectus. Pellentesque nec nam aliquam sem et tortor consequat id porta. Sapien et ligula ullamcorper malesuada. Fringilla ut morbi tincidunt augue interdum velit euismod. Vitae turpis massa sed elementum tempus egestas sed. Netus et malesuada fames ac turpis egestas integer eget aliquet. Blandit cursus risus at ultrices mi tempus imperdiet nulla malesuada. Aliquam eleifend mi in nulla posuere sollicitudin aliquam ultrices sagittis. Nisi est sit amet facilisis. Vitae congue eu consequat ac felis. Facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nisl condimentum id venenatis a condimentum. Rutrum tellus pellentesque eu tincidunt tortor. Semper auctor neque vitae tempus quam. Pellentesque adipiscing commodo elit at imperdiet dui accumsan sit. Mi ipsum faucibus vitae aliquet nec ullamcorper sit. Enim lobortis scelerisque fermentum dui faucibus. Volutpat ac tincidunt vitae semper quis lectus nulla at. Pellentesque id nibh tortor id aliquet lectus proin. Blandit cursus risus at ultrices mi. Iaculis eu non diam phasellus vestibulum lorem. 
        </p>
      </div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="4"
      >
        <h1>
          Handling status: suspended
        </h1>
        <p>
          Paragraph 4. Paragraph 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id nibh tortor id aliquet. Sed lectus vestibulum mattis ullamcorper velit sed. Nec nam aliquam sem et tortor. Elit eget gravida cum sociis natoque. Ipsum a arcu cursus vitae. Massa sapien faucibus et molestie ac feugiat sed. Turpis egestas maecenas pharetra convallis posuere morbi leo. Duis at consectetur lorem donec massa sapien faucibus et molestie. Purus sit amet luctus venenatis lectus. Pellentesque nec nam aliquam sem et tortor consequat id porta. Sapien et ligula ullamcorper malesuada. Fringilla ut morbi tincidunt augue interdum velit euismod. Vitae turpis massa sed elementum tempus egestas sed. Netus et malesuada fames ac turpis egestas integer eget aliquet. Blandit cursus risus at ultrices mi tempus imperdiet nulla malesuada. Aliquam eleifend mi in nulla posuere sollicitudin aliquam ultrices sagittis. Nisi est sit amet facilisis. Vitae congue eu consequat ac felis. Facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nisl condimentum id venenatis a condimentum. Rutrum tellus pellentesque eu tincidunt tortor. Semper auctor neque vitae tempus quam. Pellentesque adipiscing commodo elit at imperdiet dui accumsan sit. Mi ipsum faucibus vitae aliquet nec ullamcorper sit. Enim lobortis scelerisque fermentum dui faucibus. Volutpat ac tincidunt vitae semper quis lectus nulla at. Pellentesque id nibh tortor id aliquet lectus proin. Blandit cursus risus at ultrices mi. Iaculis eu non diam phasellus vestibulum lorem. 
        </p>
      </div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="5"
      >
        <h1>
          Determining verification documents needed
        </h1>
        <p>
          Paragraph 5. ellus vestibulum lorem. 
        </p>
      </div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="6"
      >
        <h1>
          Level 2 Heading
        </h1>
        <p>
          Paragraph 6. Paragraph 1. Lorem ipsum dolor sicididun diam phasellus vestibulum lorem. 
        </p>
      </div>
      <div
        className="css-4d756m-Content egzsi4z2"
        id="7"
      >
        <h1>
          Handling retry status
        </h1>
        <p>
          Paragraph  7. Paragraph 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id nibh tortor id aliquet. Sed lectus vestibulum mattis ullamcorper velit sed. Nec nam aliquam sem et tortor. Elit eget gravida cum sociis natoque. Ipsum a arcu cursus vitae. Massa sapien faucibus et molestie ac feugiat sed. Turpis egestas maecenas pharetra convallis posuere morbi leo. Duis at consectetur lorem donec massa sapien faucibus et molestie. Purus sit amet luctus venenatis lectus. Pellentesque nec nam aliquam sem et tortor consequat id porta. Sapien et ligula ullamcorper malesuada. Fringilla ut morbi tincidunt augue interdum velit euismod. Vitae turpis massa sed elementum tempus egestas sed. Netus et malesuada fames ac turpis egestas integer eget aliquet. Blandit cursus risus at ultrices mi tempus imperdiet nulla malesuada. Aliquam eleifend mi in nulla posuere sollicitudin aliquam ultrices sagittis. Nisi est sit amet facilisis. Vitae congue eu consequat ac felis. Facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nisl condimentum id venenatis a condimentum. Rutrum tellus pellentesque eu tincidunt tortor. Semper auctor neque vitae tempus quam. Pellentesque adipiscing commodo elit at imperdiet dui accumsan sit. Mi ipsum faucibus vitae aliquet nec ullamcorper sit. Enim lobortis scelerisque fermentum dui faucibus. Volutpat ac tincidunt vitae semper quis lectus nulla at. Pellentesque id nibh tortor id aliquet lectus proin. Blandit cursus risus at ultrices mi. Iaculis eu non diam phasellus vestibulum lorem. 
        </p>
      </div>
    </div>
    <div
      className="css-rgn0gs-StickyNav egzsi4z1"
    >
      <div
        className="css-a2c0yj-Group e1ycoeu80"
      >
        <div
          className="css-1w28r9v-Heading e1ycoeu81"
        >
          ON THIS PAGE
        </div>
        <a
          className="tab-1"
          onClick={[Function]}
        >
          The Basics
        </a>
        <a
          className="tab-2"
          onClick={[Function]}
        >
          Verification statuses and corresponding events
        </a>
        <a
          className="tab-3"
          onClick={[Function]}
        >
          Handling retry status
        </a>
        <a
          className="tab-3"
          onClick={[Function]}
        >
          Handling status: suspended
        </a>
        <a
          className="tab-4"
          onClick={[Function]}
        >
          Determining verification documents needed
        </a>
        <a
          className="tab-2"
          onClick={[Function]}
        >
          Level 2 Heading
        </a>
        <a
          className="tab-3"
          onClick={[Function]}
        >
          Handling retry status
        </a>
        <div
          className="css-1txt6v-PageTop e1ycoeu82"
          onClick={[Function]}
        >
          Top of Page
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1oxk9ln-Footer egzsi4z3"
  />
</div>
`;

exports[`default 1`] = `
<div
  className="css-a2c0yj-Group e1ycoeu80"
>
  <div
    className="css-1w28r9v-Heading e1ycoeu81"
  >
    ON THIS PAGE
  </div>
  <a
    className="tab-1"
    onClick={[Function]}
  >
    The Basics
  </a>
  <a
    className="tab-2"
    onClick={[Function]}
  >
    Verification statuses and corresponding events
  </a>
  <a
    className="tab-3"
    onClick={[Function]}
  >
    Handling retry status
  </a>
  <a
    className="tab-3"
    onClick={[Function]}
  >
    Handling status: suspended
  </a>
  <a
    className="tab-4"
    onClick={[Function]}
  >
    Determining verification documents needed
  </a>
  <a
    className="tab-2"
    onClick={[Function]}
  >
    Level 2 Heading
  </a>
  <a
    className="tab-3"
    onClick={[Function]}
  >
    Handling retry status
  </a>
  <div
    className="css-1txt6v-PageTop e1ycoeu82"
    onClick={[Function]}
  >
    Top of Page
  </div>
</div>
`;
